src/EnergyPlus/WaterManager.cc

```diff
@@ -1080,6 +1085,11 @@ namespace WaterManager {
         }
 
         state.dataWaterData->WaterStorage(TankNum).ThisTimeStepVolume = state.dataWaterData->WaterStorage(TankNum).LastTimeStepVolume + NetVolAdd;
+        if (state.dataWaterData->WaterStorage(TankNum).ThisTimeStepVolume >= state.dataWaterData->WaterStorage(TankNum).ValveOffCapacity) {
+            state.dataWaterData->WaterStorage(TankNum).LastTimeStepFilling = false;
+        }
+
```

Comment:
At the end of the iteration, you check whether you are above or equal to `ValveOffCapacity`, in which case you switch `LastTimeStepFilling` to false.